reframe/frontend/executors/policies.py

```diff
@@ -55,6 +59,21 @@ def runcase(self, case):
         except BaseException:
             task.fail(sys.exc_info())
         finally:
+
+            # Execute cleanup of dependencies if all dependent cases have been
+            # executed
+            for dep in self.dependency_tree[case]:
+                self.dependency_count[dep] -= 1
+                if self.dependency_count[dep] == 0:
+                    # Check if dep has failed before cleaning
+                    for t in self.stats.tasks():
```

Comment:
It would be better to associate the reference counter to the `RegressionTask` so that you avoid this for loop here and the use of `stats`.